### PR TITLE
ColorMapper: Fix uninitialized return value

### PR DESCRIPTION
**Pull request by uklotzde**
Detected by GCC 10

### CODE REVIEW

File: src/controllers/colormapper.cpp [hunk 1]
```diff
@@ -31,32 +31,51 @@ double colorDistance(QRgb a, QRgb b) {
...
+        m_cache.insert(desiredColor, nearestColor);
+        return nearestColor;
     }
 
-    kLogger.trace()
-            << "ColorMapper found matching color for" << desiredColor << ":"
-            << "Color =" << nearestColor;
-    m_cache.insert(desiredColor, nearestColor);
-    return nearestColor;
+    // Not found
```
Comment (Holzhaus) [inline src/controllers/colormapper.cpp:72]: Do we have a `DEBUG_ASSERT_NOT_REACHED`? Otherwise just use `DEBUG_ASSERT(false)` or something like that. Not finding a suitable color should never happen unless `m_availableColors` is empty (which should be guarded by debug assertions as well).

File: src/controllers/colormapper.cpp [hunk 2]
```diff
@@ -31,32 +31,51 @@ double colorDistance(QRgb a, QRgb b) {
...
+    if (kLogger.traceEnabled()) {
```
Comment (Holzhaus) [inline src/controllers/colormapper.cpp:73]: OT: Can we check the log level for the logger inside `trace()` and early exit without printing something? Maybe have a global singleton that each logger can ask for its log level (DEBUG by default) and we can override the level for each logger via the command line? That would be handy to prevent log spam of unrelated stuff when you develop and test a particular feature.

Comment (uklotzde) [reply src/controllers/colormapper.cpp:73]: I don't think so. Afaik operator<< is not evaluated lazily.

Comment (uklotzde) [reply src/controllers/colormapper.cpp]: We need to revisit the logging at some time. The `Logger` emerged during Qt4 times. Only a very limited workaround for the lack of a serious logging framework.

File: src/controllers/colormapper.cpp [hunk 3]
```diff
@@ -31,32 +31,47 @@ double colorDistance(QRgb a, QRgb b) {
...
-    if (i != m_cache.constEnd()) {
-        DEBUG_ASSERT(m_availableColors.contains(i.value()));
-        kLogger.trace()
-                << "ColorMapper cache hit for" << desiredColor << ":"
-                << "Color =" << i.value();
-        return i.value();
+    const auto iCachedColor = m_cache.constFind(desiredColor);
+    if (iCachedColor != m_cache.constEnd()) {
+        const QRgb cachedColor = iCachedColor.key();
+        DEBUG_ASSERT(m_availableColors.contains(cachedColor));
```
Comment (Holzhaus) [inline src/controllers/colormapper.cpp:37]: Just tested, i'm hitting this debug assertion.

Comment (Holzhaus) [reply src/controllers/colormapper.cpp:37]: I think it should be iCachedColor.value(), not key()

**Review by Holzhaus — CHANGES_REQUESTED** [review]

File: src/controllers/colormapper.cpp (hunk 3 above)
Comment (uklotzde) [reply src/controllers/colormapper.cpp:37]: Tricky due to the same types. We should have test for this thing. But at least the debug assertion revealed it early.

**Review by Holzhaus — APPROVED** [review]
LGTM, thank you very much.
